### ISSUE TITLE
[Production] [0004] Saga: cross the build FOB Rubicon

### ISSUE THREAD

**Issue by mwpowellhtx** [mwpowellhtx/KP-Liberation#23]
### Basic Information
**Mission version:** `0.98.dev-s2`
**Map used:** _Altis_
**Mods used:** _..._

### Short Issue Description
What do we mean by that, re: _Rubicon_. We think the pattern _building FOB_ can potentially be generalized to serve at least both purposes. Abstracting out things like _originating position_, _build radius_, _classname_, etc. While at the same time, if we're careful about it, resolving the issue with _direction_ and _up vectors_ failing to cross the wire somehow somewhere during the process.

### Expected Behaviour
As with _building FOB_, plus resolving the _direction_ and _up vectors_ failing to cross over the wire during the process. We think that this should work in a very similar manner _building a storage container_ at a _factory sector_.

### Steps to Reproduce the Issue
Re: _direction_ and _up vectors_, _FOB building_, for instance, positions "okay" during the build staging. When acknowledged, the _up vector_ seems to get ignored, or was never relayed properly in the first place, we think. There is a lot of client/server, event handling, etc, going on, so the indirection is massive. But we think that's what is happening.

### Overall objectives
- [x] We think that perhaps the _build FOB_ sequence of events might be a decent place to start; objectively, if we can leverage the approach in both places, that would be ideal, but we will not set out with that as the primary objective in mind.
- [x] Also remember, now with production _manager dialog_ being presentable, including a _working sector map_, also include markers corresponding to _storage actual positions_.
- [x] Another thing to consider is how to _align storage objects with their respective factory sector_. In terms of _object serialization_, we think that we can possibly leverage the _serialized variables_ approach to indicate which _storage_ belongs with which _factory sector_. This is especially important, as well, should such sectors potentially _overlap_, geospatially speaking.
- [x] While we're in there considering the _FOB building_ approach, we should also gain some comprehension concerning _build geometry_ and the issues dealing ith _direction indicators_ while arranging objects in the build queue, appropriate _up vectors_ upon accepting the build queue, and so forth.

**Comment by mwpowellhtx** [comment]
- [ ] We think that perhaps the _build FOB_ sequence of events might be a decent place to start; objectively, if we can leverage the approach in both places, that would be ideal, but we will not set out with that as the primary objective in mind.
- [ ] Also remember, now with production _manager dialog_ being presentable, including a _working sector map_, also include markers corresponding to _storage actual positions_.
- [ ] Another thing to consider is how to _align storage objects with their respective factory sector_. In terms of _object serialization_, we think that we can possibly leverage the _serialized variables_ approach to indicate which _storage_ belongs with which _factory sector_. This is especially important, as well, should such sectors potentially _overlap_, geospatially speaking.

**Comment by mwpowellhtx** [comment]
Pretty much completed. We resolved the indicated observations. We did somewhat refactor things in the building of _factory sector storage_, but that is contained to _factory sectors_ only at the moment. Approaching an opportunity such as this, we need to be careful of all the callbacks, _server to server_, _client to client_, and especially _crossing client server boundaries_, and which parameters and other arguments might be available.

This is especially critical, because in the A3 engine, the moment when we need those bits, in the context of a running callback, let's say, may have already fallen well out of scope and vanished. That's a problem. So we get a bit creative stashing some bits, either on the player him or herself, in a namespace possibly, along these lines.
